ElementX/Sources/Screens/RoomScreen/View/RoomScreen.swift

```diff
@@ -49,15 +49,11 @@ struct RoomScreen: View {
                     .environment(\.roomContext, context)
             }
             .overlay(alignment: .top) {
-                if context.viewState.isPinningEnabled {
-                    // TODO: Implement tapping logic + hiding when scrolling
-                    PinnedItemsBannerView(pinIndex: 1,
-                                          pinsCount: 3,
-                                          pinContent: .init(stringLiteral: "Content"),
-                                          onMainButtonTap: { },
-                                          onViewAllButtonTap: { })
+                if context.viewState.shouldShowPinBanner {
+                    pinnedItemsBanner
                 }
             }
+            .animation(.elementDefault, value: context.viewState.shouldShowPinBanner)
```

Comment:
Worth double checking this (might want to be a ZStack or something instead), but this would make sure to isolate the animations to only the banner. (It's a bit sad that while we now have `if` expressions you can't put the modifier directly on the `if` but oh well)
```suggestion
                Group {
                    if context.viewState.shouldShowPinBanner {
                        pinnedItemsBanner
                    }
                }
                .animation(.elementDefault, value: context.viewState.shouldShowPinBanner)
            }
```